dialogue-core/src/main/java/com/palantir/dialogue/core/BlacklistingChannel.java

```diff
@@ -49,7 +50,7 @@
  * unblacklisted. Without this functionality, hundreds of requests could be sent to a still-broken
  * server before the first of them returns and tells us it's still broken.
  */
-final class BlacklistingChannel implements LimitedChannel {
+final class BlacklistingChannel implements CompositeLimitedChannel {
```

Comment:
can we tweak the names so that your CompositeLimitedChannel just becomes LimitedChannel and then rename LimitedChannel -> NodeSelectionChannel (because that's our PinUntilError, RoundRobin implementations)?